### PR TITLE
[Experimental] Fix Decoding of Array of Empty Elements 

### PR DESCRIPTION
**Pull request by bwetherfield**

### CODE REVIEW

File: Sources/XMLCoder/Decoder/XMLKeyedDecodingContainer.swift [hunk 1]
```diff
@@ -282,6 +272,13 @@ extension XMLKeyedDecodingContainer {
             return empty
         }
 
+        // If we are looking at a coding key value intrinsic where the expected type is `String` and
+        // the value is empty, return `""`.
+        if strategy(key) != .attribute, elements.isEmpty, attributes.isEmpty, type == String.self,
+            key.stringValue == "value" || key.stringValue == "" {
+            return "" as! T
```
[the remark this answers is not in the document]
Comment (bwetherfield) [reply Sources/XMLCoder/Decoder/XMLKeyedDecodingContainer.swift]: @jsbean , can you think of a way of avoiding the force cast here?

[the remark this answers is not in the document]
Comment (jsbean) [reply Sources/XMLCoder/Decoder/XMLKeyedDecodingContainer.swift]: @bwetherfield, sorry I just saw this! Hmm, I think it must be.

This is not my proudest moment 😬, but I can't think of another way. Perhaps @MaxDesiatov has thoughts!

[the remark this answers is not in the document]
Comment (bwetherfield) [reply Sources/XMLCoder/Decoder/XMLKeyedDecodingContainer.swift]: Fixed!

**Comment by MaxDesiatov** [comment]
@bwetherfield is this one going to be resubmitted in some other form?

**Comment by bwetherfield** [comment]
Yes, I was just trying to clarify the order here by closing, since it consumes the changes of the PR I just submitted. Trying to separate out the functionality as best I can!

**Pull request closed without merging**